### PR TITLE
Migrate from `gumdrop` to `clap`

### LINKED ISSUE #72: Migrate to `clap`
When I wrote the first version of this tool, `clap` was basically unmaintained and `gumdrop` was a nice small CLI handler. Nowadays, `gumdrop` is unmaintained and `clap` has seen significant improvement, so we should swap to it.

### PR DESCRIPTION
**Pull request by str4d**
Closes #72.

Best reviewed by commit.

### CODE REVIEW

File: src/cli.rs
Comment (nuttycom) [inline src/cli.rs:1]: This module name doesn't make sense to me; it should be called `wallet` - the whole tool is a CLI.

Comment (str4d) [reply src/cli.rs:1]: That module defines the _entire_ CLI, for all commands.

File: src/cli.rs [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+}
+
+fn parse_currency(data: &str) -> Result<Currency, String> {
+    Currency::from_code(data).ok_or_else(|| format!("Invalid currency '{data}'"))
+}
+
+#[derive(Debug, Parser)]
+pub(crate) struct MyOptions {
+    /// Path to the wallet directory
+    pub(crate) wallet_dir: Option<String>,
```
Comment (str4d) [inline src/cli.rs:15]: I should have moved this in the last commit. Gonna figure out how to do that cleanly now.

**Review by nuttycom — DISMISSED** [review]
utACK 3c9df4cd035b5bfda52ce118e2eb19aaba89ca58

**Review by nuttycom — APPROVED** [review]
re-utACK b15eaebc6d2c490b6f35b20041bc9f64e095062a

**Pull request merged**